controllers/drpolicy_controller.go

```diff
@@ -378,5 +387,67 @@ func (u *objectUpdater) finalizerRemove() error {
 func (r *DRPolicyReconciler) SetupWithManager(mgr ctrl.Manager) error {
 	return ctrl.NewControllerManagedBy(mgr).
 		For(&ramen.DRPolicy{}).
+		Watches(
+			&source.Kind{Type: &corev1.ConfigMap{}},
+			handler.EnqueueRequestsFromMapFunc(r.configMapMapFunc),
+			builder.WithPredicates(predicate.ResourceVersionChangedPredicate{}),
+		).
+		Watches(
+			&source.Kind{Type: &corev1.Secret{}},
+			handler.EnqueueRequestsFromMapFunc(r.secretMapFunc),
+			builder.WithPredicates(predicate.ResourceVersionChangedPredicate{}),
+		).
 		Complete(r)
 }
+
+func (r *DRPolicyReconciler) configMapMapFunc(configMap client.Object) []reconcile.Request {
+	if configMap.GetName() != HubOperatorConfigMapName || configMap.GetNamespace() != NamespaceName() {
+		return []reconcile.Request{}
+	}
+
+	drpolicies := &ramen.DRPolicyList{}
+	if err := r.Client.List(context.TODO(), drpolicies); err != nil {
+		return []reconcile.Request{}
+	}
+
+	requests := make([]reconcile.Request, len(drpolicies.Items))
+	for i, drpolicy := range drpolicies.Items {
+		requests[i].Name = drpolicy.GetName()
+	}
+
+	return requests
+}
+
+func (r *DRPolicyReconciler) secretMapFunc(secret client.Object) (requests []reconcile.Request) {
+	drpolicies := &ramen.DRPolicyList{}
+	if err := r.Client.List(context.TODO(), drpolicies); err != nil {
+		return
+	}
+
+	_, ramenConfig, err := ConfigMapGet(context.TODO(), r.APIReader)
+	if err != nil {
+		return
+	}
+
+	for _, drpolicy := range drpolicies.Items {
+		for _, drCluster := range drpolicy.Spec.DRClusterSet {
+			s3Profile := RamenConfigS3StoreProfilePointerGet(ramenConfig, drCluster.S3ProfileName)
+			if s3Profile == nil {
+				continue
+			}
+
+			if s3Profile.S3SecretRef.Namespace == secret.GetNamespace() &&
+				s3Profile.S3SecretRef.Name == secret.GetName() {
+				requests = append(requests,
+					reconcile.Request{
+						NamespacedName: types.NamespacedName{
+							Name: drpolicy.GetName(),
+						},
+					},
+				)
```

Comment:
i think control should continue to the outer for loop here to avoid extra reconciles when a policy's clusters share the same s3 secret

Reply:
`break` was added here